src/components/Approval/Approval.tsx

```diff
@@ -157,6 +146,38 @@ export const Approval = () => {
     showErrorToast,
   ])
 
+  const toggleIsExactAllowance = useCallback(() => {
+    setIsExactAllowance(!isExactAllowance)
+  }, [isExactAllowance])
+
+  useEffect(() => {
+    setBuildApprovalTxInput(undefined)
+    createBuildApprovalTxInput(isExactAllowance)
+      .then(buildApprovalTxInput => {
+        setBuildApprovalTxInput(buildApprovalTxInput)
+      })
+      .catch(e => {
+        showErrorToast(e)
+        return history.push(TradeRoutePaths.Input)
+      })
+  }, [isExactAllowance, createBuildApprovalTxInput, history, showErrorToast])
+
+  const approvalNetworkFeeCryptoHuman = useMemo(() => {
+    if (!buildApprovalTxInput) return
+
+    const { gasLimit, gasPrice, maxFeePerGas } = buildApprovalTxInput
+
+    // use EIP-1551 gas where possible
+    const totalGasPrice = maxFeePerGas ?? gasPrice
+
+    if (!totalGasPrice) return
+
+    return baseUnitToHuman({
+      value: bn(gasLimit).times(totalGasPrice),
+      inputExponent: sellFeeAsset?.precision ?? 0,
+    })
```

Comment:
unsure on correctness of this, appreciate comments

Reply:
good catch!

Reply:
this is where we would need to add the optimism l1 fee to the displayed transaction fee, or we could simply pass back `txFee` from `getFeesFromContractData`, though this does not take into consideration eip1559 support for the most accurate representation of fees to be paid

Reply:
This makes more sense. I've pushed a change to handle display fee for optimism. The tricky part was that the need to build the tx to know what the gas limit was so to avoid re-computing everything and risk values drifting based on market conditions its now all calculated within the same function and separated in the return type.